src/topology/constructions.lean

```diff
@@ -904,6 +904,14 @@ lemma closure_subtype {x : {a // p a}} {s : set {a // p a}}:
   x ∈ closure s ↔ (x : α) ∈ closure ((coe : _ → α) '' s) :=
 closure_induced
 
+lemma dense_range_of_subset_closure (s : set α) :
+  dense_range (λ (x : s), (⟨x, subset_closure x.2⟩ : closure s)) :=
```

Comment:
This lemma is not my favorite (i.e., the statement is ugly). Are you not able to use `dense_range_iff_closure_range` where you apply this lemma (with some finagling)? If you really need this lemma, at the very least it should be stated with `subtype.map`.

Reply:
Thanks for tour comments and golf! 

I do agree that this is not a pretty lemma. I need in the following setting: `K` is a `subfield  : F` and I have a ring hom `f : K →+* F`. I want to extend `f` to a ring homomorphism from `closure K` to `F`. For that I need to prove that the inclusion of `K` into `closure K` is `dense_range`, hence the lemma. I do not mind deleting the lemma and proving the fact directly where I need it.

Reply:
In that case, I would say you should either:

1. prove this locally when you need it (especially if you only need it once)
2. prove a more general result in place of this lemma for `subtype.map`.

What I mean by (2) above is something like:

```lean
lemma subtype.map_dense_range_iff {α β : Type*} [topological_space β] {p : α → Prop}
  {q : β → Prop} (f : α → β) (h : ∀ (a : α), p a → q (f a)) :
  dense_range (subtype.map f h) ↔ {b | q b} ⊆ closure (f '' {a | p a}) := sorry
```

Reply:
Well, (2) sounds a lot more fun and a great opportunity to learn something new so I'll try that next week when I have more time :)

Reply:
Using the above statement gives a scary error: 
```
kernel failed to type check declaration 'subtype.map_dense_range_iff'
this is usually due to a buggy tactic or a bug in the builtin elaborator
elaborated type:
  ∀ {α : Type u_1} {β : Type u_2} [_inst_2 : topological_space β] {p : α → Prop} {q : β → Prop}
  (f : α → β),
    (α → β) →
    ∀ (h : ∀ (a : α), p a → q (f a)),
      dense_range (subtype.map f h) ↔ {b : β | q b} ⊆ closure (f '' {a : α | p a})
elaborated value:
  λ {α : Type u_1} {β : Type u_2} [_inst_2 : topological_space β] {p : α → Prop} {q : β → Prop}
  (f g : α → β) (h : ∀ (a : α), p a → q (f a)),
    {mp := λ (h : dense_range (subtype.map f h)), sorry, mpr := sorry}
nested exception message:
type mismatch at application
  @dense_range (@subtype β q) (@subtype.topological_space β q _inst_2) {a // p a}
term
  {a // p a}
has type
  Type u_1
but is expected to have type
  Type u_3
```

Reply:
So it turns out finally that I didn't need this lemma since [dense_embedding.subtype_emb](https://leanprover-community.github.io/mathlib_docs/topology/dense_embedding.html#dense_embedding.subtype_emb) is doing what I want. Just needed to figure out how to use it ;)
Anyway, what I need now is just the innocuous `dense_embedding_id`.